jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriter.java

```diff
@@ -209,6 +227,17 @@ public void doWrite(Server server, Query query, Iterable<Result> results) throws
 			resultAttribute.addTo(resultTagMap, result);
 		}
 
+		if (typeNamesAsTags) {
+			log.debug("adding typeNames as tags");
+			Map<String, String> typeNameValueMap = TypeNameValue.extractMap(resultTagMap.get("typeName"));
+			log.debug("this is typeNameValueMap {} ==== {} ", typeNameValueMap, this.typeNames);
+			for (String typeToTag : this.typeNames) {
+				if (typeNameValueMap.get(typeToTag) != null) {
```

Comment:
The point of using `get()` was to remove one lookup into the map, so you'll need to store that result and not do the same `get()` again on the next line :)